stats.go

```diff
@@ -438,6 +568,44 @@ func newHistogramBucket(
 	return bucket
 }
 
+type metricExpiryTracker struct {
+	lastUpdateUnix int64
+	expired        uint32
+	scope          *scope
+	name           string
+}
+
+// expiredAndUpdated checks the metric is expired and if so
+// will return true only for the first routine that updates
+// an expired metric, signalling to insert the metric back
+// into the parent scope.
+func (m *metricExpiryTracker) expiredAndUpdated() bool {
+	// We perform a LoadUint32 first for performance reasons
+	if atomic.LoadUint32(&m.expired) == 1 && m.scope != nil &&
+		atomic.CompareAndSwapUint32(&m.expired, 1, 0) {
+		return true
```

Comment:
we always call `resetUpdatedTime` in the caller when this returns `true` right? should we just call `resetUpdatedTime` here so callers don't need to